app/services/map_data_service.py

```diff
@@ -0,0 +1,288 @@
+            
+            "rating": result.get("rating", 0.0),
+            "reviews": result.get("reviews", []),
+            
+            "price_level": result.get("price_level", 0),
+            "tags": result.get("types", []),
+            "wheelchair_accessible_entrance": result.get("wheelchair_accessible_entrance", False),
+            
+            "serves_beer": result.get("serves_beer", False),
+            "serves_breakfast": result.get("serves_breakfast", False),
+            "serves_brunch": result.get("serves_brunch", False),
+            "serves_dinner": result.get("serves_dinner", False),
+            "serves_lunch": result.get("serves_lunch", False),
+            "serves_vegetarian_food": result.get("serves_vegetarian_food", False),
+            "serves_wine": result.get("serves_wine", False),
+            "takeout": result.get("takeout", False),
+        }
+        
+        current_opening_hours = result.get("current_opening_hours")
+        if current_opening_hours:
+            processed_data["opening_hours"] = self.get_opening_hours(current_opening_hours)
+        
+        return processed_data
+    
+    def process_from_json(self, json_file_path: str) -> Dict[str, Any]:
+        with open(json_file_path, 'r', encoding='utf-8') as f:
+            place_data = json.load(f)
+        
+        return self.process_place_data(place_data)
+    
+    def save_processed_data(self, processed_data: Dict[str, Any], output_path: str) -> None:
+        serializable_data = {}
+        
+        for key, value in processed_data.items():
+            if isinstance(value, (OpeningHours, Coordinates)):
+                serializable_data[key] = value.__dict__
+            else:
+                serializable_data[key] = value
+        
+        with open(output_path, 'w', encoding='utf-8') as f:
+            json.dump(serializable_data, f, indent=2, ensure_ascii=False)
+        
+        print(f"✅ Processed data saved to {output_path}")
+
+    def _extract_address_components(self, place: Dict[str, Any]) -> Address:
+        address_components = place.get("result", {}).get("address_components", [])
+        extracted = {}
+        
+        for component in address_components:
+            component_types = component.get("types", [])
+            long_name = component.get("long_name", "")
+            
+            for target_type in self.target_address_types:
+                if target_type in component_types:
+                    extracted[target_type] = long_name
+                    break
+        
+        return Address(
+            postal_code=extracted.get("postal_code", ""),
+            administrative_area_level_1=extracted.get("administrative_area_level_1", ""),
+            administrative_area_level_2=extracted.get("administrative_area_level_2", ""),
+            sublocality_level_1=extracted.get("sublocality_level_1", ""),
+            formatted_address=place.get("result", {}).get("formatted_address", "")
+        )
```

Comment:
The method is almost identical to the address extraction logic in `process_place_data`.
Suggest refactoring to share code and avoid duplication.